website/src/utils/planner.ts

```diff
@@ -134,3 +163,15 @@ export function getTotalMC(
   // Remove nulls using .filter(Boolean)
   return sum(modules.map(getModuleCredit).filter(Boolean));
 }
+
+/**
+ * Returns if a planner module is year-long
+ */
+export function isYearLong(module: Pick<PlannerModuleInfo, 'moduleInfo' | 'customInfo'>): boolean {
+  return module.moduleInfo ? Boolean(module.moduleInfo.attributes?.year) : false;
+}
+
+export function isSemester(plannerSemester: PlannerModuleSemester | string) { 
```

Comment:
This makes the function a type guard which means you won't need to cast https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards 

```suggestion
export function isSemester(plannerSemester: PlannerModuleSemester): plannerSemester is Semester { 
```